### PR TITLE
Maven config launchable from IntelliJ IDEA

### PR DESCRIPTION
**Pull request by imikushin**
Changed some `provided` scopes to `runtime`. 
Your IntelliJ IDEA enjoy! :)

### CODE REVIEW

**Comment by ibuildthecloud** [comment]
Fyi @hibooboo2

File: code/packaging/app-parent/pom.xml [hunk 1]
```diff
@@ -110,7 +110,7 @@
                         </goals>
                         <configuration>
                             <includeGroupIds>org.eclipse.jetty</includeGroupIds>
-                            <scope>provided</scope>
+                            <!--<scope>provided</scope>-->
```
Comment (thaJeztah) [inline code/packaging/app-parent/pom.xml]: Might want to remove this instead of commenting out?

File: code/packaging/app-parent/pom.xml [hunk 2]
```diff
@@ -123,7 +123,7 @@
                         <configuration>
                             <includeGroupIds>io.cattle</includeGroupIds>
                             <includeArtifactIds>cattle-launcher</includeArtifactIds>
-                            <scope>provided</scope>
+                            <!--<scope>provided</scope>-->
```
Comment (thaJeztah) [inline code/packaging/app-parent/pom.xml]: Same here

File: code/packaging/app-parent/pom.xml (hunk 1 above)
Comment (imikushin) [inline code/packaging/app-parent/pom.xml]: yeah, you're right! forgot it

**Comment by imikushin** [comment]
Hope the build will pass CI without those commented lines as well.

**Comment by ibuildthecloud** [comment]
As I mentioned before, the maven setup is quite complicated and fragile.  With these changes we end up adding to WEB-INF/lib of the cattle war the following

```
WEB-INF/lib: cattle-framework-launcher-0.5.0-SNAPSHOT.jar
WEB-INF/lib: com.sun.el-2.2.0.v201108011116.jar
WEB-INF/lib: javax.el-2.2.0.v201108011116.jar
WEB-INF/lib: javax.servlet-3.0.0.v201112011016.jar
WEB-INF/lib: javax.servlet.jsp-2.2.0.v201112011158.jar
WEB-INF/lib: javax.servlet.jsp.jstl-1.2.0.v201105211821.jar
WEB-INF/lib: jetty-client-8.1.11.v20130520.jar
WEB-INF/lib: jetty-continuation-8.1.11.v20130520.jar
WEB-INF/lib: jetty-http-8.1.11.v20130520.jar
WEB-INF/lib: jetty-io-8.1.11.v20130520.jar
WEB-INF/lib: jetty-jmx-8.1.11.v20130520.jar
WEB-INF/lib: jetty-jsp-8.1.11.v20130520.jar
WEB-INF/lib: jetty-security-8.1.11.v20130520.jar
WEB-INF/lib: jetty-server-8.1.11.v20130520.jar
WEB-INF/lib: jetty-servlet-8.1.11.v20130520.jar
WEB-INF/lib: jetty-servlets-8.1.11.v20130520.jar
WEB-INF/lib: jetty-util-8.1.11.v20130520.jar
WEB-INF/lib: jetty-webapp-8.1.11.v20130520.jar
WEB-INF/lib: jetty-websocket-8.1.11.v20130520.jar
WEB-INF/lib: jetty-xml-8.1.11.v20130520.jar
WEB-INF/lib: org.apache.jasper.glassfish-2.2.2.v201112011158.jar
WEB-INF/lib: org.apache.taglibs.standard.glassfish-1.2.0.v201112081803.jar
WEB-INF/lib: org.eclipse.jdt.core-3.7.1.jar
```

All of those files should not be in the WEB-INF/lib because they are Jetty or Servlet libs.  We need to find an approach that doesn't add these files

**Comment by ibuildthecloud** [comment]
One approach could be to create a new project like "cattle-app-intellij" that has "cattle-app" as the parent but then overrides these scopes.  That may cause the mvn build to take twice as long maybe

**Comment by imikushin** [comment]
@ibuildthecloud I reverted the scopes back to provided. 

I've found a way to run cattle in intellij without adding a new subproject - use Maven:

Open 'Maven Projects'. Find 'cattle-app' -> Plugins -> jetty -> jetty:run. Create `cattle-app [jetty:run]`. Customize to your taste (e.g. add JVM parameters). Run or Debug.

The app looks working pretty much all right. 

I've run integration tests against it, and only got 3 of them fail (the last time I tried, a different set of tests failed on a seemingly clear setup, but anyway): 
`test_account_delete`
`test_short_lived_container`
`test_container_fields`

**Comment by ibuildthecloud** [comment]
I'll try this out, in eclipse I've always avoided calling maven because 1) it's slows the startup 2) classpath issues.

Can you confirm that your setup is using the local changes from you IDE and not what is installed in ~/.m2

**Comment by imikushin** [comment]
Definitely. There's a checkbox for that: 'Resolve Workspace artifacts' (in the Run configuration edit). 
I inserted a logging message in the source and it showed in the log.

**Comment by hibooboo2** [comment]
Excited about this. Thanks @imikushin

**Comment by imikushin** [comment]
You are very welcome! Please let me know how this works. If all is well, I can add some wiki edits to describe IntelliJ setup.

**Comment by imikushin** [comment]
So, did anyone try to run/debug cattle in IntelliJ (as [described above](https://github.com/rancherio/cattle/pull/333#issuecomment-90024046)) yet? How did it work?

**Comment by ibuildthecloud** [comment]
This didn't work for me.  I think maven/intellij is gettings confused between workspace artifacts and what is in ~/.m2 repo.  It's running an odd mix of old code.

**Comment by ibuildthecloud** [comment]
FWIW, I don't launch maven through eclipse either because of crap like that.  I just prefer to call the main class and normal java program.  So I liked the approach better of changing the scopes that intellij will just have the right classpath and I can just run a java program (not maven project, yuck).

**Comment by imikushin** [comment]
@ibuildthecloud did you check 'Resolve Workspace artifacts' (in the Run configuration edit)?

**Comment by ibuildthecloud** [comment]
Yes I did.

**Comment by imikushin** [comment]
@ibuildthecloud @hibooboo2 
Please test: Run/Debug `io.cattle.platform.launcher.Main` within the context
(workdir and classpath) of `cattle-app-intellij` module

**Comment by imikushin** [comment]
Note: I've rebased on top of the latest master

**Comment by ibuildthecloud** [comment]
This is working for me!!!  I'll keep using intellij over the next week.  If all is good I'll merge.  Small changes though:

1. I don't know why host-stats is in app/pom.xml.  You can move it to app-parent/pom.xml and then not duplicate it in intellij
2. I moved app-intellij to its own profile so that the default mvn build does not include it, but then you can just easily activate the profile in intellij.

In cattle/pom.xml

```xml
       <profile>
            <id>intellij</id>
            <modules>
                <module>code/packaging/app-intellij</module>
            </modules>
        </profile>
```

**Comment by imikushin** [comment]
Hey, you don't need another profile: the new module barely adds anything to build time (around 0.09s). Of course, it's easy to enable a profile in IntelliJ, but why the extra steps and more code? Non-IntelliJ people can just ignore it: the module's artifact (now just another pom.xml) doesn't get copied into the final jar or affect their runtime environment in any way (though, I suspect, it can also be used to run the thing in Eclipse).

**Comment by ibuildthecloud** [comment]
@imikushin Oh yeah, your right I just assumed the packaging was war like app/pom.xml.  I'll just test that it doesn't have any side effects on the wars/jars produced.  I can't imagine it would.

File: code/packaging/app-parent/pom.xml [hunk 3]
```diff
@@ -110,7 +110,6 @@
                         </goals>
                         <configuration>
                             <includeGroupIds>org.eclipse.jetty</includeGroupIds>
-                            <scope>provided</scope>
```
Comment (ibuildthecloud) [inline code/packaging/app-parent/pom.xml]: Seems like this is going to have side effects.  Let me test it out

Comment (imikushin) [inline code/packaging/app-parent/pom.xml]: IntelliJ reported `<scope>` was not supposed to be here

Comment (ibuildthecloud) [inline code/packaging/app-parent/pom.xml]: Your right, this is a typo.

File: code/packaging/app-intellij/src [hunk 4]
```diff
@@ -0,0 +1 @@
+../app/src
```
Comment (ibuildthecloud) [inline code/packaging/app-intellij/src]: Can you remove this?

**Comment by ibuildthecloud** [comment]
This has been working great for me so far.  I'd like to merge.  Can you remove the `../app/src`?  I'm not sure what that is.  symlink?

**Comment by imikushin** [comment]
I sure can! But I guess I better not.
Even if the `Main` process would launch much quicker:
```
09:26:13.448 [main] INFO  org.eclipse.jetty.server.Server - jetty-8.1.11.v20130520
09:26:13.981 [main] INFO  o.e.jetty.server.AbstractConnector - Started SelectChannelConnector@0.0.0.0:8080
09:26:13.981 [main] INFO  ConsoleStatus - [DONE ] [1962ms] Startup Succeeded, Listening on port 8080
```
Yep. That's the entire process log without that symlink :)

**Comment by ibuildthecloud** [comment]
Oh, thats because you need to set the working directory to code/packaging/app.  I'm trying to figure out how we can share the launch configuration like we do with Eclipse.

**Pull request merged**